src/poetry/utils/vcs/git.py

```diff
@@ -0,0 +1,193 @@
+from __future__ import annotations
+
+import re
+import shutil
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from dulwich import porcelain
+from dulwich.client import get_transport_and_path
+from dulwich.config import ConfigFile
+from dulwich.config import parse_submodules
+from dulwich.errors import NotGitRepository
+from dulwich.refs import ANNOTATED_TAG_SUFFIX
+from dulwich.repo import Repo
+from poetry.core.utils.helpers import canonicalize_name
+
+
+if TYPE_CHECKING:
+    from dulwich.client import FetchPackResult
+    from dulwich.client import GitClient
+
+
+class Git:
```

Comment:
This class doesn't handle any states. So it would be more pythonic to not create a class here and just have the methods within the `git` module.

Reply:
Not necessarily. This pattern allows for a more cleaner usage. Don't think this is required even if no state is maintained.